Gems/GraphModel/Code/Source/Model/Slot.cpp

```diff
@@ -107,171 +121,36 @@ namespace GraphModel
-            name,
-            displayName,
-            DataTypeList{ dataType },
-            defaultValue,
-            description,
-            extendableSlotConfiguration,
-            enumValues,
-            visibleOnNode,
-            editableOnNode);
-    }
-
-    SlotDefinitionPtr SlotDefinition::CreateInputData(
-        AZStd::string_view name,
-        AZStd::string_view displayName,
-        DataTypeList supportedDataTypes,
-        AZStd::any defaultValue,
-        AZStd::string_view description,
-        ExtendableSlotConfiguration* extendableSlotConfiguration,
-        const AZStd::vector<AZStd::string>& enumValues,
-        bool visibleOnNode,
-        bool editableOnNode)
-    {
-        AZStd::shared_ptr<SlotDefinition> slotDefinition = AZStd::make_shared<SlotDefinition>();
-        slotDefinition->m_slotDirection = SlotDirection::Input;
-        slotDefinition->m_slotType = SlotType::Data;
-        slotDefinition->m_name = name;
-        slotDefinition->m_displayName = displayName;
-        slotDefinition->m_supportedDataTypes = supportedDataTypes;
-        slotDefinition->m_defaultValue = defaultValue;
-        slotDefinition->m_description = description;
-        slotDefinition->m_enumValues = enumValues;
-        slotDefinition->m_visibleOnNode = visibleOnNode;
-        slotDefinition->m_editableOnNode = editableOnNode;
-
-        HandleExtendableSlotRegistration(slotDefinition, extendableSlotConfiguration);
-
-        return slotDefinition;
-    }
-
-    SlotDefinitionPtr SlotDefinition::CreateOutputData(
-        AZStd::string_view name,
-        AZStd::string_view displayName,
-        DataTypePtr dataType,
-        AZStd::string_view description,
-        ExtendableSlotConfiguration* extendableSlotConfiguration,
+    SlotDefinition::SlotDefinition(
+        SlotDirection slotDirection,
+        SlotType slotType,
+        const AZStd::string& name,
+        const AZStd::string& displayName,
+        const AZStd::string& description,
+        const DataTypeList& supportedDataTypes,
+        const AZStd::any& defaultValue,
+        int minimumSlots,
+        int maximumSlots,
+        const AZStd::string& addButtonLabel,
+        const AZStd::string& addButtonTooltip,
         const AZStd::vector<AZStd::string>& enumValues,
         bool visibleOnNode,
         bool editableOnNode)
+        : m_slotDirection(slotDirection)
+        , m_slotType(slotType)
+        , m_name(name)
+        , m_displayName(displayName)
+        , m_description(description)
+        , m_supportedDataTypes(supportedDataTypes)
+        , m_defaultValue(defaultValue)
+        , m_minimumSlots(AZStd::max(0, AZStd::min(minimumSlots, maximumSlots)))
+        , m_maximumSlots(AZStd::max(0, AZStd::max(minimumSlots, maximumSlots)))
+        , m_addButtonLabel(addButtonLabel)
+        , m_addButtonTooltip(addButtonTooltip)
+        , m_enumValues(enumValues)
+        , m_visibleOnNode(visibleOnNode)
+        , m_editableOnNode(editableOnNode)
```

Comment:
All the string, vector and any types can be moved into the their members.
```suggestion
        : m_slotDirection(slotDirection)
        , m_slotType(slotType)
        , m_name(AZStd::move(name))
        , m_displayName(AZStd::move(displayName))
        , m_description(AZStd::move(description))
        , m_supportedDataTypes(AZStd::move(supportedDataTypes))
        , m_defaultValue(AZStd::move(defaultValue))
        , m_minimumSlots(AZStd::max(0, AZStd::min(minimumSlots, maximumSlots)))
        , m_maximumSlots(AZStd::max(0, AZStd::max(minimumSlots, maximumSlots)))
        , m_addButtonLabel(AZStd::move(addButtonLabel))
        , m_addButtonTooltip(AZStd::move(addButtonTooltip))
        , m_enumValues(AZStd::move(enumValues))
        , m_visibleOnNode(visibleOnNode)
        , m_editableOnNode(editableOnNode)
```